File: src/main/java/com/example/PojoValidator.java
package com.example;

import jakarta.validation.Validation;
import jakarta.validation.Validator;
import jakarta.validation.ValidatorFactory;

public class PojoValidator {
	private final static ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
	private final static Validator validator = factory.getValidator();

	public static <T> void validate(T pojo) {
		try {
			final var violations = validator.validate(pojo);
			if (violations.size() > 0) {
				System.out.println("POJO " + pojo + "\ncontains errors: ");
				violations.forEach(u -> System.out.println("  '" + u.getPropertyPath().toString() + "'" + " " + u.getMessage()));
				System.out.println("****************************************\n");
			}
		} catch(Exception e) {
			e.printStackTrace();
		}
	}
}

File: src/main/kotlin/com/example/kotlin/PojoValidator.kt
package com.example.kotlin

import jakarta.validation.ConstraintViolation
import jakarta.validation.Validation
import java.util.function.Consumer

object PojoValidator {
    private val factory = Validation.buildDefaultValidatorFactory()
    private val validator = factory.validator
    fun <T> validate(pojo: T) {
        try {
            val violations = validator.validate(pojo)
            if (violations.size > 0) {
                println("POJO $pojo\ncontains errors: ")
                violations.forEach(Consumer { u: ConstraintViolation<T> ->
                    println(
                        "  '" + u.propertyPath.toString() + "'" + " " + u.message
                    )
                })
                println("****************************************\n")
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
